eox_nelp/openedx_filters/xapi/filters.py

```diff
@@ -126,32 +130,32 @@ def run_filter(self, transformer, result):  # pylint: disable=arguments-differ
 
         Arguments:
             transformer <XApiTransformer>: Transformer instance.
-            result <Activity>: Object activity for events related to enrollments.
+            result <Activity>: Module or Problem object activity for any kind of event.
 
         Returns:
             Activity: Modified activity.
         """
-        # Following line is hard-coded since this logic has not been tested with other events.
-        if transformer.event["name"] != "edx.grades.problem.submitted":
-            return {
-                "result": result
-            }
+        if result.definition.type in [constants.XAPI_ACTIVITY_QUESTION, constants.XAPI_ACTIVITY_MODULE]:
+            display_name = transformer.get_data('display_name')
 
-        display_name = transformer.get_data('display_name')
+            # Get label from module descriptor block.
+            usage_id = transformer.get_data("data.problem_id") or transformer.get_data("data.block_id")
+            usage_key = UsageKey.from_string(usage_id)
+            item = modulestore().get_item(usage_key)
+            label = get_item_label(item)
 
-        # Get label from module descriptor block.
-        usage_key = UsageKey.from_string(transformer.get_data("data.problem_id"))
-        item = modulestore().get_item(usage_key)
-        label = get_item_label(item)
+            # Get course languge block.
+            course = get_course_from_id(transformer.get_data('course_id'))
+            course_language = course.get("language")
```

Comment:
I think that we could create a function for language if this piece of code is multiple used.
https://github.com/eduNEXT/eox-nelp/pull/120/files#diff-7f85cc618f32f4937416b9a541239a453597fac05c0552e3c34011f6faa0aaadR100-R104

Reply:
IMO this change makes more sense in a different PR, [here](https://github.com/eduNEXT/eox-nelp/pull/120) you mentioned the necessity of using EN_US, so I think this change should be included in that pr